scarlet/initialization.py

```diff
@@ -237,6 +228,42 @@ def init_extended_source(
         )
         morph = prox_monotonic(morph, 0).reshape(morph.shape)
 
+    return morph, bg_rms, center_index
+
+
+def init_extended_source(
+    sky_coord,
+    frame,
+    observations,
+    coadd=None,
+    coadd_rms=None,
+    thresh=1,
+    symmetric=True,
+    monotonic="flat",
+    min_grad=0.1,
+):
+    """Initialize the source that is symmetric and monotonic
+    See `ExtendedSource` for a description of the parameters
+    """
+    if not hasattr(observations, "__iter__"):
+        observations = (observations,)
+
+    # determine initial SED from peak position
+    # SED in the frame for source detection
+    spectrums = [get_psf_sed(sky_coord, obs, frame) for obs in observations]
+    spectrum = np.concatenate(spectrums).reshape(-1)
+
+    morph, bg_rms, center_index = morph_init(sky_coord,
+                                             frame,
+                                             observations,
+                                             spectrum,
```

Comment:
I'm quite sure that this needs to be `spectra` because you need the amplitudes in the same order as `observation` and not concatenated. That's the whole reason for calling `get_psf_sed` separately for all observations above.

Reply:
Not relevant anymore.